### ISSUE TITLE
[KEYBOARD]: EYB - Focus MUST be returned to the trigger when modals are closed in IE11

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#9241]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️  Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

IE11 doesn't return focus properly to the `<button>` that triggers the modal window when users close a few of the modals. I identified at least two that are not returning focus correctly. This issue is specific to IE11, so could be identified with a console.error check. Screenshot attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] Focus is returned properly to the Learn more buttons that open modal windows when they are closed
- [ ] Screen readers follow this logic and return focus to the correct button also

## Environment

* Windows 10
* IE11
* JAWS

## Steps to Recreate

1. Start JAWS screen reader in Internet Explorer 11
1. Load one of the relevant profile views
1. Click on Learn more next to the School Calendar label
1. Close the modal window
1. Verify the focus is returned to the top of the page by pressing `DOWN_ARROW` and listening to what is read aloud

## Screenshots or Trace Logs

![Screenshot (35)](https://user-images.githubusercontent.com/934879/82359799-0423f700-99ce-11ea-8f12-724853ca0a67.png)

---
![Screen Shot 2020-05-18 at 9 08 32 PM](https://user-images.githubusercontent.com/934879/82277157-cf239000-994c-11ea-90a1-f6564aeaa8f0.png)

**Comment by 1Copenut** [comment]
@desireeturner I'm going to remove this one from the Staging Accessibility Review - GIBCT - Section 103 Updates epic because it's an Estimate Your Benefits issue. I got the two conflated last night while auditing. I'll update the tags so we can track effectively.

/cc @delli-gatti-michael && @sonntag-adam

**Comment by hughes-dustin** [comment]
Unable to reproduce in IE11 using JAWS on current master branch.

**Comment by 1Copenut** [comment]
@hughes-dustin I'm still able to re-create this issue on staging. I just tested the URL staging.va.gov/gi-bill-comparison-tool/profile/31000627 (Midland University, Fremont, NE) using IE11 + JAWS 2019. LMK if you'd like to set up a time to video chat and we can review together.

**Comment by hughes-dustin** [comment]
@1Copenut  we might have to do that. I stepped through all the modals on that link and couldn't get the focus to not return back to the button. I should also point out that I'm using JAWS 2020.

**Comment by delli-gatti-michael** [comment]
I just attempted to recreate this one using JAWS 2020 and it looks like closing a modal by pressing enter/spacebar while focused on the actual "close this modal" button works as intended with the focus returning to the link that opened the modal in the first place. However, if a user closes a modal using the esc key, then the behavior described in the defect above happens. 

I've tried several different modals, including the ones pictured above, and this seems to be consistent throughout the Comparison Tool

**Comment by hughes-dustin** [comment]
I was unable to recreate the defect using the escape key as pointed out by @delli-gatti-michael. We might be dealing with a nondeterministic issue. Which is most unfortunate.

**Comment by delli-gatti-michael** [comment]
After restarting my Windows VM and trying again, I can't reproduce it with JAWS 2020 and IE either.

**Comment by hughes-dustin** [comment]
branch: bah-9241
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/12912

**Comment by hughes-dustin** [comment]
On hold until Trevor Pierce gets input on JAWS 2019 vs JAWS 2020.

**Comment by 1Copenut** [comment]
@hughes-dustin The VA 508 office got back to me on this one and said they're using a mix of JAWS 2019 and 2020. I'm not able to get myself moved up to JAWS 2020 yet, so I consider this one still live and valid.

I'll go ahead and make my recommendation to the VA 508 office when I submit Section 103 for audit later today that this is a known issue and we're working on a fix.

/cc @desireeturner

**Comment by hughes-dustin** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/13273
Branch: bah-9241-d

**Comment by delli-gatti-michael** [comment]
On the `bah-9241-d` branch, the only way I can reproduce this is by highlighting a link that will open a modal and rapidly spamming the enter key effectively hitting the link and then the close modal button over and over.  I don't think that's going to actually happen in real life, so I'm going to approve the PR.

**Comment by hughes-dustin** [comment]
> On the `bah-9241-d` branch, the only way I can reproduce this is by highlighting a link that will open a modal and rapidly spamming the enter key effectively hitting the link and then the close modal button over and over. I don't think that's going to actually happen in real life, so I'm going to approve the PR.

I was able to reproduce it through _vigorous_ spamming of the enter key. I believe this is related to what @delli-gatti-michael  stated earlier.

**Comment by delli-gatti-michael** [comment]
It's probably worth noting that this is only with the enter key. Pressing the spacebar, for example, will not reproduce the defect.

**Comment by 1Copenut** [comment]
I confirmed things are behaving about as well as can be expected of IE11 and focus is being returned to the button that opens modals. I'll move this one to close.
/cc @delli-gatti-michael
